cmd/log-shuttle/internal/determine_test.go

```diff
@@ -0,0 +1,53 @@
+package internal
+
+import "testing"
+
+func TestDetermineAWSRegion(t *testing.T) {
+	for _, tc := range []struct {
+		input    string
+		expected string
+		err      bool
+	}{
+		{input: "foo.bar.com", expected: "", err: true},
+		{input: "", expected: "", err: true},
+		{input: "logs.foo.amazonaws.com", expected: "foo", err: false},
+		{input: "logs.us-east-2.amazonaws.com", expected: "us-east-2", err: false},
+		{input: "logs.us-east-2.amazonaws.com.cn", expected: "us-east-2", err: false},
```

Comment:
What happens if someone targets an EC2 node like: `ec2-54-203-136-119.us-west-2.compute.amazonaws.com`? Would the "failure to determine the service so we default to the logplex formatter" behavior work?

Reply:
It would. The host name you provided isn't a valid ec2 service hostname.

Reply:
I added a test to prove it.